src/adapters/node.ts

```diff
@@ -214,7 +214,11 @@ async function sendResponse(socket: Duplex, res: Response) {
       socket.write(chunk);
     }
   }
+  socket.end();
```

Comment:
Perhaps similar to Node.js [impl](https://github.com/nodejs/node/blob/196efd77f59e377fff1dde38d0f4132e4a998732/lib/net.js#L802-L810) you shared, we could check on `writable`, and `writableFinished` flags?

I'm not sure `finish` event is guaranteed to always be called and it could mean we might have situations with never resolving promise

Reply:
Good point. I'll update it to check on the flags. The Node.js docs says the `finish` event will be called after `stream.end()` so if we're calling `end` conditionally, the event might not be called.